pages/api/auth/[...nextauth].js

```diff
@@ -1,92 +1,156 @@
+            // Note: This option is ignored if using JSON Web Tokens
+            // updateAge: 24 * 60 * 60, // 24 hours
+        },
+
+        // JSON Web tokens are only used for sessions if the `jwt: true` session
+        // option is set - or by default if no database is specified.
+        // https://next-auth.js.org/configuration/options#jwt
+        jwt: {
+            // A secret to use for key generation (you should set this explicitly)
+            // secret: 'INp8IvdIyeMcoGAgFGoA61DdBglwwSqnXJZkgz8PSnw',
+            // Set to true to use encryption (default: false)
+            // encryption: true,
+            // You can define your own encode/decode functions for signing and encryption
+            // if you want to override the default behaviour.
+            // encode: async ({ secret, token, maxAge }) => {},
+            // decode: async ({ secret, token, maxAge }) => {},
+        },
+
+        // You can define custom pages to override the built-in pages.
+        // The routes shown here are the default URLs that will be used when a custom
+        // pages is not specified for that route.
+        // https://next-auth.js.org/configuration/pages
+        pages: {
+            // signIn: '/api/auth/signin',  // Displays signin buttons
+            // signOut: '/api/auth/signout', // Displays form with sign out button
+            // error: '/api/auth/error', // Error code passed in query string as ?error=
+            // verifyRequest: '/api/auth/verify-request', // Used for check email page
+            // newUser: null // If set, new users will be directed here on first sign in
+        },
+
+        // Callbacks are asynchronous functions you can use to control what happens
+        // when an action is performed.
+        // https://next-auth.js.org/configuration/callbacks
+        callbacks: {
+            jwt: async (token, user, account, profile, isNewUser) => {
+                // True only on first sign in
+                if (isNewUser) {
+                    let uuid = req.cookies.uuid
+                    if (!uuid) {
+                        uuid = uuidv4()
+                    }
+                    token.id = uuid
+                    await fetch(
+                        `${process.env.NEXTAUTH_URL}/api/saveUserUuid`,
+                        {
+                            method: 'POST',
+                            headers: {
+                                'content-type': 'application/json',
+                                accept: 'application/json',
+                            },
+                            body: JSON.stringify({
+                                user_id: user.id,
+                                uuid: uuid,
+                            }),
+                        },
+                    )
+                }
+                // On other sign ins
+                else if (user) {
+                    const session_uuid = req.cookies.uuid
+                    if (session_uuid) {
```

Comment:
Shouldn't you check that session_uuid is different from the existing uuid of the existing uuid before triggering a merge? They may be the same, and the merge should not be needed.

Reply:
The design is so that this would never happen. Won’t hurt to check, though. So I’ll add it.

Reply:
Addressed in ecd08e3